rest/api.go

```diff
@@ -324,8 +331,14 @@ func (h *handler) handleHeapProfiling() error {
 	if err != nil {
 		return err
 	}
-	pprof.WriteHeapProfile(f)
-	f.Close()
+	err = pprof.WriteHeapProfile(f)
+	if err != nil {
+		return err
```

Comment:
I expect we still need to close the file before returning in this error case?

Reply:
Fixed in latest revision